rasa/core/policies/rule_policy.py

```diff
@@ -286,11 +286,11 @@ def _check_slots_fingerprint(
     ) -> Set[Text]:
         expected_slots = set(fingerprint.get(SLOTS, {}))
         current_slots = set(state.get(SLOTS, {}).keys())
-        if expected_slots == current_slots:
-            # all expected slots are satisfied
+        if expected_slots.issubset(current_slots):
+            # all expected slots are found in the currently set slots
             return set()
 
-        return expected_slots
+        return expected_slots.difference(current_slots)
```

Comment:
@Ghostvv what do you think, shall I change this similarly in `_check_active_loops_fingerprint` too? Also, I've noticed that there we use `expected_active_loops` but this will always be at most one loop, or not?

Reply:
yes in the other case I think it can be only one loop

Reply:
oh, effectively in case of loops, I check if it a subset

Reply:
I think, you can leave it as is

Reply:
aaah

Reply:
one action can set 2 loops: `some_loop` and `null`, that's why it is expected loops

Reply:
Ah, right, now it makes sense to me. Thanks! I'll leave everything as it is then.